src/components/form/GitAuthenticationForm.vue

```diff
@@ -0,0 +1,89 @@
+<template>
+  <q-form
+    @submit="onSubmit"
+    data-cy="git-authentication-form"
+    class="q-gutter-md git-form"
+  >
+    <q-input
+      filled
+      v-model="username"
+      :label="$t('page.modelizer.settings.gitAuthentication.username')"
+      data-cy="git-username-input"
+    />
+    <q-input
+      filled
+      v-model="token"
+      :label="$t('page.modelizer.settings.gitAuthentication.token')"
+      data-cy="git-token-input"
+    />
+    <div class="flex row items-center justify-center">
+      <q-btn
+        icon="fa-solid fa-save"
+        :label="$t('actions.default.save')"
+        type="submit"
+        :loading="submitting"
+        data-cy="git-form-submit"
+        color="positive">
+        <template v-slot:loading>
+          <q-spinner-dots/>
+        </template>
+      </q-btn>
+    </div>
+  </q-form>
+</template>
+
+<script setup>
+import { Notify } from 'quasar';
+import { useI18n } from 'vue-i18n';
+import { ref } from 'vue';
+import {
+  getProjectById,
+  saveProject,
+} from 'src/composables/Project';
+import GitEvent from 'src/composables/events/GitEvent';
+
+const emit = defineEmits(['project-git:save']);
+
+const props = defineProps({
+  projectName: {
+    type: String,
+    required: true,
+  },
+});
+
+const { t } = useI18n();
+const project = getProjectById(props.projectName);
+const username = ref(project.git?.username);
+const token = ref(project.git?.token);
+const submitting = ref(false);
+
+/**
+ * Save git project, emit events and send positive toast.
```

Comment:
`GitAddRemote` calls `gitAddRemote(project)` then `saveProject(project)` to save it in localStorage
Here, `GitAuthentication` only calls `saveProject(project)` since we just need to add `username` and `token` to the git object.
I changed the doc for both files to be more precise